node/src/bootstrap/bootstrap_attempt.rs

```diff
@@ -73,17 +71,23 @@ pub(crate) struct BootstrapAttempt {
     pub started: AtomicBool,
     pub stopped: AtomicBool,
     pub frontiers_received: AtomicBool,
+    bootstrap_started_observer: Arc<Mutex<Vec<Box<dyn Fn(String, String) + Send + Sync>>>>,
+    bootstrap_ended_observer:
+        Arc<Mutex<Vec<Box<dyn Fn(String, String, String, String) + Send + Sync>>>>,
 }
 
 impl BootstrapAttempt {
     pub fn new(
-        websocket_server: Option<Arc<WebsocketListener>>,
         block_processor: Weak<BlockProcessor>,
         bootstrap_initiator: Weak<BootstrapInitiator>,
         ledger: Arc<Ledger>,
         id: String,
         mode: BootstrapMode,
         incremental_id: u64,
+        bootstrap_started_observer: Arc<Mutex<Vec<Box<dyn Fn(String, String) + Send + Sync>>>>,
+        bootstrap_ended_observer: Arc<
+            Mutex<Vec<Box<dyn Fn(String, String, String, String) + Send + Sync>>>,
+        >,
```

Comment:
It isn't clear what the arguments to the callback are. It takes 4 strings. What does it mean? Let's create a struct here that contains the data for the callback.
Why are block count, duration and mode passed as strings? We should use the underlying data type instead